spec/virtual_total_spec.rb

```diff
@@ -16,16 +16,34 @@
           .to eq([author0, author1, author2].map(&:id))
       end
 
-      it "calculates totals locally" do
-        author0_id = Author.create_with_books(0).id
-        author2_id = Author.create_with_books(2).id
+      it "calculates totals using a query" do
+        author0 = Author.create_with_books(0).reload
+        author2 = Author.create_with_books(2).reload
```

Comment:
Is the `.reload` here just to make sure we are in a fresh state, and then we are only doing the `total_books` query instead of including the `.find` here?

If so, makes sense, just making sure I am understanding the change correctly.

Reply:
I was worried about the creation in `create_with_books` leaving some of the associations pre populated/loaded

Reply:
I like it, and I liked that you droped the `.find` in the `.expect`.  Just making sure I was reading it right.  👍